File: frontend/src/model/user/student.ts
import {User} from "./user";
import {Promo} from "../promo";

export class Student extends User {
  override promoId: number;

  promo: Promo | undefined;

  constructor(firstName: string, lastName: string,
              email: string, password: string,
              promoId: number,
              studentId?: number | undefined,
              token?: string | undefined,
              pfpImgPath?: string | undefined,
              pfpImgUrl?: string | undefined) {
    super(firstName, lastName, email, password, studentId, token, pfpImgPath, pfpImgUrl);
    this.promoId = promoId;
    this.studentId = studentId;
  }

  static override fromJson(jsonStudent: Student): Student {
    let student = new Student(jsonStudent.firstName, jsonStudent.lastName,
      jsonStudent.email, jsonStudent.password, jsonStudent.promoId, jsonStudent.studentId,
      jsonStudent.token, jsonStudent.pfpImgPath,
      jsonStudent.pfpImgUrl);
    if(jsonStudent.promo != undefined) student.promo = Promo.fromJson(jsonStudent.promo);

    return student;
  }
}

File: backend/src/main/java/arbremojo/backend/model/user/Student.java
package arbremojo.backend.model.user;

import arbremojo.backend.model.Promo;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "student")
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Student extends User {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "student_id", nullable = false)
    private Integer studentId;
    @Column(name = "first_name", nullable = false)
    private String firstName;
    @Column(name = "last_name", nullable = false)
    private String lastName;
    @Column(name = "email", nullable = false)
    private String email;
    @Column(name = "password", nullable = false)
    private String password;
    @Column(name = "token")
    private String token;
    @Column(name = "pfp_img_path")
    private String pfpImgPath;

    @Column(name = "promo_id", nullable = false)
    private Integer promoId;

    @ManyToOne(fetch = FetchType.EAGER, targetEntity = Promo.class)
    @JoinColumn(name = "promo_id", insertable = false, updatable = false)
    private Promo promo;

    @Override
    public int getUserId() {
        return studentId;
    }
}
